#################################################################################
#   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.          #
#                                                                               #
#   Licensed under the Apache License, Version 2.0 (the "License").             #
#   You may not use this file except in compliance with the License.            #
#   You may obtain a copy of the License at                                     #
#                                                                               #
#       http://www.apache.org/licenses/LICENSE-2.0                              #
#                                                                               #
#   Unless required by applicable law or agreed to in writing, software         #
#   distributed under the License is distributed on an "AS IS" BASIS,           #
#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.    #
#   See the License for the specific language governing permissions and         #
#   limitations under the License.                                              #
#################################################################################

# Publish all joint states -----------------------------------
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 60

left_rear_wheel_velocity_controller:
  # joints:
  #   - left_rear_wheel_joint
  # type: velocity_controllers/JointGroupVelocityController
  type: velocity_controllers/JointVelocityController
  joint: left_rear_wheel_joint
  pid: {p: 1.0, i: 0.0, d: 0.0, i_clamp: 0.0}

right_rear_wheel_velocity_controller:
  # joints:
  #   - right_rear_wheel_joint
  # type: velocity_controllers/JointGroupVelocityController
  type: velocity_controllers/JointVelocityController
  joint: right_rear_wheel_joint
  pid: {p: 1.0, i: 0.0, d: 0.0, i_clamp: 0.0}


left_front_wheel_velocity_controller:
  # joints:
  #   - left_front_wheel_joint
  # type: velocity_controllers/JointGroupVelocityController
  type: velocity_controllers/JointVelocityController
  joint: left_front_wheel_joint
  pid: {p: 0.5, i: 0.0, d: 0.0, i_clamp: 0.0}


right_front_wheel_velocity_controller:
  # joints:
  #   - right_front_wheel_joint
  # type: velocity_controllers/JointGroupVelocityController
  type: velocity_controllers/JointVelocityController
  joint: right_front_wheel_joint
  pid: {p: 0.5, i: 0.0, d: 0.0, i_clamp: 0.0}


left_steering_hinge_position_controller:
  # joints:
  #   - left_steering_hinge_joint
  # type: position_controllers/JointGroupPositionController
  type: position_controllers/JointPositionController
  joint: left_steering_hinge_joint
  pid: {p: 1.0, i: 0.0, d: 0.5}


right_steering_hinge_position_controller:
  # joints:
  #   - right_steering_hinge_joint
  # type: position_controllers/JointGroupPositionController
  type: position_controllers/JointPositionController
  joint: right_steering_hinge_joint
  pid: {p: 1.0, i: 0.0, d: 0.5}

gazebo_ros_control/pid_gains:
  left_rear_wheel_joint: {p: 0.1, i: 0.0, d: 0.0, i_clamp: 0.0}
  right_rear_wheel_joint: {p: 0.1, i: 0.0, d: 0.0, i_clamp: 0.0}
  left_front_wheel_joint: {p: 0.1, i: 0.0, d: 0.0, i_clamp: 0.0}
  right_front_wheel_joint: {p: 0.1, i: 0.0, d: 0.0, i_clamp: 0.0}
  left_steering_hinge_joint: {p: 0.05, i: 0.0, d: 0.0}
  right_steering_hinge_joint: {p: 0.05, i: 0.0, d: 0.0}